src/components/cl/hier/allgatherv/allgatherv.c

```diff
@@ -69,53 +69,48 @@ static inline ucc_status_t find_leader_rank(ucc_base_team_t *team,
    dst buffer is contiguous */
 static inline ucc_status_t is_block_ordered(ucc_cl_hier_team_t *cl_team, int *ordered)
 {
-    ucc_topo_t *topo = cl_team->super.super.params.team->topo;
-    ucc_sbgp_t *sbgp;
-    int         is_block_ordered;
+    ucc_topo_t  *topo             = cl_team->super.super.params.team->topo;
+    ucc_sbgp_t  *all_nodes        = NULL;
+    int          is_block_ordered = 1;
+    int          n_nodes;
+    ucc_status_t status;
+    ucc_rank_t   prev_rank, curr_rank;
+    ucc_rank_t   i, j;
 
     if (cl_team->is_block_ordered != -1) {
         is_block_ordered = cl_team->is_block_ordered;
     } else {
-        sbgp = ucc_topo_get_sbgp(topo, UCC_SBGP_FULL_HOST_ORDERED);
-        is_block_ordered = ucc_ep_map_is_identity(&sbgp->map) ? 1 : 0;
-        cl_team->is_block_ordered = is_block_ordered;
-    }
-
-    *ordered = is_block_ordered;
-
-    return UCC_OK;
-}
-
-/* Node leader subgroup is always ordered by ascending host_id. If the team's ranks are
-   not in the same order, then node leader subgroup allgatherv won't be enough, we'll
-   have to use a staging buffer and unpack to reorder each leader's contribution.
-   So, this func will check that the team ranks in the ldr sbgp are ascending */
-static inline ucc_status_t is_host_ordered(ucc_cl_hier_team_t *cl_team, int *ordered)
```

Comment:
Here I remove is_host_ordered. It checks that the node leader subgroup is in order with the team's ranks (the comment actually has it backwards). This function isn't needed anymore though because I sort the leader ranks to be in order of the team during sbgp creation.